src/dialogs/TileDBPromptOptionsWidget.ts

```diff
@@ -79,13 +81,27 @@ export class TileDBPromptOptionsWidget extends Widget {
     owner_label.textContent = 'Owner:';
     const owner_input = document.createElement('select');
 
-    options.owners.forEach((owner) => {
-      const option = document.createElement('option');
-      option.setAttribute('value', owner);
-      option.setAttribute('label', owner);
-      owner_input.append(option);
-    });
+    addOptionsToSelectInput(owner_input, options.owners, options.selectedOwner);
+
     owner_input.setAttribute('name', 'owner');
+
+    owner_input.onchange = async (e): Promise<any> => {
+      const newOwner = (e.currentTarget as HTMLSelectElement).value;
+      // Reset credentials input
+      resetSelectInput(s3_cred_selectinput);
+      const tileDBAPI = await getTileDBAPI();
+      const credentialsResponse = await tileDBAPI.checkAWSAccessCredentials(
+        newOwner
+      );
+      const newCredentials = credentialsResponse.data;
+      const credentials: string[] = newCredentials.map((cred) => cred.name);
+      addOptionsToSelectInput(
+        s3_cred_selectinput,
+        credentials,
+        options.defaultS3CredentialName
```

Comment:
Is this the `defaultS3CredentialName` from the new owner (aka the organization) or the user's default still?

Reply:
It's from the user, if it won't find that value it will fallback to the first credential. I'm adding this now to get organization's default s3 credential name if the owner is not the user.

Reply:
Thanks for the adjustment